File: backend/app/domain/models/Prompt.scala
package domain.models

/**
 * プロンプト
 * @param mainPrompt メインプロンプト
 * @param subPrompt サブプロンプト
 * @param option オプション
 * @param isComparable 比較可能か
 * @param isNumeric 数値か
 * @param unit 単位
 */
case class Prompt(
    val mainPrompt: String,
    val subPrompt: String,
    val option: Option[String],
    val isComparable: Boolean,
    val isNumeric: Boolean,
    val unit: Option[String]
)

File: frontend/src/types/Prompt.ts
/**
 * 質問の型
 * @param prompts 質問のリスト
 * @param isNumeric 数値かどうか
 * @param isComparable 比較可能かどうか
 * @param unit 単位
 * @param questionName 質問の名前
 */
export type Prompt = {
  prompts: Array<string>;
  isNumeric: boolean;
  isComparable: boolean;
  unit: string;
  questionName: string;
};
